testkit-java/src/main/scala/com/akkaserverless/javasdk/testkit/impl/TestKitActorSystem.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.testkit.impl
+
+import akka.actor.ActorSystem
+
+/**
+ * INTERNAL API
+ */
+private[akkaserverless] object TestKitActorSystem {
+  // Only used to materialize streams (expected to be short lived) in user logic, same system re-used across tests is fine
+  // FIXME leaks one actor system per test run though, because no callback to terminate, that's not great
+  lazy val system = ActorSystem(s"testkit-actor-system-${System.currentTimeMillis()}")
```

Comment:
The only other option I could think of is to let the concrete testkit classes lazy-create and own the actor system, problem is the testkit also does not have a lifecycle, so would require adding that further complicating things.

Not allowing unit tests to materializing the incoming stream does not feel like an option.
Open for any good ideas.

Reply:
I don't see how we could do that differently. Since we generate the test skeleton, we can add a teardown method and stop the testkit with a good chunk of comments explaining the reasons. 

If users touch that method then yes, they will be leaking.

Reply:
Decided to go back on this for now and revisit supporting it in the testkit separately as that is a bigger task involving changes to how the testkit is interacted with from the tests (both for in progress Scala testkits and the Java ones).

Reply:
yes, let's take that later